smartsim/ml/tf/data.py

```diff
@@ -80,17 +70,80 @@ def __data_generation(self, indices):
 
         return x, y
 
+    def __getitem__(self, index):
+        if len(self) < 1:
+            msg = "Not enough samples in generator for one batch. "
+            msg += "Please run init_samples() or initialize generator with init_samples=True"
+            raise ValueError(msg)
+        # Generate indices of the batch
+        indices = self.indices[index * self.batch_size : (index + 1) * self.batch_size]
+
+        # Generate data
+        x, y = self.__data_generation(indices)
+
+        if y is not None:
+            return x, y
+        else:
+            return x
+
+    def __data_generation(self, indices):
```

Comment:
per our conversation, while we're at it, let's change the name of this method to be `_data_generation` (one underscore) both here and on the base class to avoid name-mangling and allow for this override to work.

Reply:
All done!